src/main/java/com/powsybl/openloadflow/network/AbstractLfBranch.java

```diff
@@ -205,4 +206,22 @@ public boolean isVoltageController() {
     public void setDiscreteVoltageControl(DiscreteVoltageControl discreteVoltageControl) {
         this.discreteVoltageControl = discreteVoltageControl;
     }
+
+    @Override
+    public BranchResult createBranchResult() {
+        if (getBus1() == null && getBus2() == null) {
+            return new BranchResult(getId(), 0, 0, 0, 0, 0, 0);
+        } else if (getBus1() == null) {
+            double currentScale = PerUnit.SB / getBus2().getNominalV();
+            return new BranchResult(getId(), 0, 0, 0,
+                getP2().eval() * PerUnit.SB, getQ2().eval() * PerUnit.SB, getI2().eval() * currentScale);
```

Comment:
Probably to do in another PR, but I think it would make sense to share the scaling methods in the `PerUnit` class, what do you think @annetill @geofjamg ?
Like
```java
class PerUnit {
    double scaleP(double p) { ... }
    double unscaleP(double p)  { ... }
    double scaleI(double i, double nominalV)  { ... }
    double unscaleI(double i, double nominalV)  { ... }
    ...
```

It would reduce a little the boiler plate code and possibilities for mistakes.

Reply:
Yes, great idea, per unit system must be hidden. I will make it myself I think.

Reply:
An afterthought : actually it could even be an interface, that way we could use different per-unit systems and compare their effect  !! :)  But I admit, it's not really useful in practice ...